workbench-core/datasets/src/dataSetService.ts

```diff
@@ -766,4 +789,31 @@ export class DataSetService {
     );
     return { data: { mountObject } };
   }
+
+  private async _updateNewDataSetPermissions(
+    dataset: DataSet,
+    request: CreateProvisionDatasetRequest
+  ): Promise<DataSetPermission[]> {
+    if (dataset.id) {
+      const permissions = _.isEmpty(request.permissions) ? [] : request.permissions!;
+      if (
+        !_.some(permissions, (p) => p.identity === request.authenticatedUser.id && p.identityType === 'USER')
```

Comment:
If the user explicitly provides permissions, should we be adding permission for the authenticated user?

Reply:
I thought about that and this is what I came up with. Let me reach out to Kevin and see what he thinks.

Reply:
If Kevin is fine with this approach, I'll approve